go/nvault/vault.go

```diff
@@ -193,34 +200,52 @@ func (v Vault) Lookup(key string) (string, error) {
 
 func (v Vault) Exists(key string) (bool, error) {
 	keyName := fmt.Sprintf("%s.key", key)
-	_, err := v.s3Client.HeadObject(context.TODO(), &s3.HeadObjectInput{Bucket: &v.cloudformationParams.BucketName, Key: &keyName})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```
5 seconds is quite aggressive :sweat_smile: admittedly S3 should always answer in that time but no need to be _that_ agressive IMO.